src/fenicsxconcrete/experiment.py

```diff
@@ -0,0 +1,35 @@
+from pydantic import RootModel
+from bcs import DirichletBCDefinition, InitialConditionDefinition, NeumannBCDefinition, BodyForceDefinition
+from mesh import MeshGenerator
+from material import MaterialDefinition, LinearElasticMaterial
+from pydantic.dataclasses import dataclass
+
+@dataclass(config=dict(arbitrary_types_allowed=True))
+class Experiment:
+    dirichlet_bcs: list[DirichletBCDefinition] | None
+    neumann_bcs: list[NeumannBCDefinition] | None
+    initial_conditions: list[InitialConditionDefinition] | None
+    body_forces: list[BodyForceDefinition] | None
```

Comment:
Only because of the body forces. Currently, each BC knows on which variable it should be applied, so the list of `DirichletBCDefinition` can contain bcs for any problem. I am however not sure about this design decision. Either the BC knows where it belongs because of some property, or alternatively it could know this because of its type. This would mean that for each physical problem we would introduce another BC type like `TemperatureBC`

Reply:
when you provide the dof (T, d) as additional parameter, then it is completely clear. Just have to make sure that the two have the same dimension.